### PR TITLE
Fix #352: Create ISSUE_TEMPLATE.MD

### LINKED ISSUE #352: Create issue template to guide contributors to quickly file issue
Adding an issue template to the repository enables project contributors to automatically see the template's contents in the issue form body. 

Refer more [here](https://help.github.com/articles/creating-an-issue-template-for-your-repository/)

### PR DESCRIPTION
**Pull request by brgnepal**
Fix #352 . Issue template.

Anything I missed here which should help contributors.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.MD [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<!--
+If you are reporting a new issue, make sure that we do not have any duplicates
+already open. You can ensure this by searching the issue list for this
+repository.
+For more information about reporting issues, see
+https://github.com/budhrg/vagrant-service-manager/blob/master/CONTRIBUTING.adoc#submitting-issues.
+-->
+
+**Environment details (OS, Provider(VirtualBox/Libvirt) etc.):**
```
Comment (navidshaikh) [inline .github/ISSUE_TEMPLATE.MD]: I think the OS and Provider should have two separate sections.

**Comment by brgnepal** [comment]
@navidshaikh Updated

File: .github/ISSUE_TEMPLATE.MD [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+For more information about reporting issues, see
+https://github.com/budhrg/vagrant-service-manager/blob/master/CONTRIBUTING.adoc#submitting-issues.
+-->
+
+**OS details:**
+
+
+**Provider:**
+VirtualBox and/or Libvirt
+
```
Comment (LalatenduMohanty) [inline .github/ISSUE_TEMPLATE.MD:14]: Should we add HyperV as an option? as CDK supports it

Comment (brgnepal) [inline .github/ISSUE_TEMPLATE.MD:14]: @LalatenduMohanty This one is for ADB issue template and until we support HyperV for ADB, it don't think its good idea.

Comment (LalatenduMohanty) [inline .github/ISSUE_TEMPLATE.MD:14]: @budhrg ok. But if a CDK user wants to file issue in VSM repo we will accept the issue, isn't it?

Comment (navidshaikh) [inline .github/ISSUE_TEMPLATE.MD:14]: @budhrg: rewrite it as
VirtualBox and/or Libvirt and/or HyperV (for CDK only)

**Comment by navidshaikh** [comment]
LGTM

**Comment by navidshaikh** [comment]
Works! Click https://github.com/projectatomic/vagrant-service-manager/issues/new to check.

**Comment by LalatenduMohanty** [comment]
Awesome! thanks @budhrg

**Comment by hferentschik** [comment]
Is this really a good idea? This creates a lot of noise imo and seems to be mainly applicable for bugs. For other issues you have to remove most of the content.

**Comment by brgnepal** [comment]
@hferentschik IMO, it is good idea as for a complete beginner its the way get start guide to write issue.

**Comment by hferentschik** [comment]
I rather ask for more details than dealing with incomplete information and asking for feedback, than being forced to cut away half of the template. Either when creating a new issue myself or cleaning up an issue created by a user/contributor. But let's see how it goes.

**Comment by brgnepal** [comment]
@hferentschik I think you are correct in your own point but I believe you are thinking of maintainer's point of view. 
This template is ideal for beginners or even QE people who write issue issue description which lacks formatting and not friendly for maintainers. Example [here](https://github.com/projectatomic/vagrant-service-manager/issues/257).

I myself formatted few of those issues description in the past. Hence, I came up with this idea.
If you want we could open up discussion in mailing list whether we should continue it or not.

**Comment by hferentschik** [comment]
> I think you are correct in your own point but I believe you are thinking of maintainer's point of view.

It seems to me that the maintainers do most of the work atm, right? So I'd optimize for their workflow, now for some potential contributor.

> This template is ideal for beginners or even QE people who write issue issue description which lacks formatting 

Except that the sheer amount of information int he template gets them as well confused and if you are not versed with GutHub's markdown as in the example you are referring to, you get a big mess of markdown which is even harder to clean up and understand.

Really the problem here is that we would like to capture more structured data, but cannot with GitHub Issues. The template is just a crutch at best. You want to be able to specify things conditionally where they make sense. Really you want a real issue tracker ;-) 

Using this template just gets into ones way imo.

> Example here.

Ask for clarification!?

**Pull request merged**